vmu.py

```diff
@@ -115,3 +90,86 @@ def canonical_src_string(sources):
     result = re.sub(r',', ' + ', result)
     result = re.sub(r'^(\d+\.\d+)(.*-> )(?!\d)', '\\1\\2\\1 ', result) # Duplicate release series, when needed
     return result.strip()
+
+class ParamError(Exception):
+    """Exception for errors in parameter files"""
+    pass
+
+class PackageSet(object):
+    """A class for comparing package sets as specified in VMU test parameters
+
+    Instance parameters:
+
+    label: string describing the package set, used for reporting
+    packages: list of packages as strings
+    selinux: boolean to turn on/off SELinux enforcing mode (default: False)
+    java: boolean to pre-install OSG java packages (default: True)
+
+    PackageSets are equal if their 'packages' lists are the same. If 'packages'
+    lists are equal but 'label' is not, ParamError is raised.
+
+    PackageSets are sorted by labels according to LABEL_ORDER. Unknown labels
+    are sorted alphanumerically after the known labels.
+    """
+
+    OSG_JAVA_DEFAULT = True
+    SELINUX_DEFAULT = False
+    LABEL_ORDER = ['All', 'All + GRAM', 'All + GRAM (3.2)', 'HTCondor', 'GridFTP', 'BeStMan', 'VOMS', 'GUMS']
+
+    def __init__(self, label, packages, selinux=SELINUX_DEFAULT, java=OSG_JAVA_DEFAULT):
+        if not label:
+            raise ParamError("PackageSet 'label' field cannot be blank")
+        if not packages:
+            raise ParamError("PackageSet 'packages' field must be non-empty list")
+        self.label = label
+        self.selinux = selinux
+        self.java = java
+
+        if self.java:
+            self.packages = ['java-1.7.0-openjdk-devel', 'osg-java7-compat', 'osg-java7-devel-compat'] + packages
+        else:
+            self.packages = packages
+
+    def __eq__(self, other):
+        if isinstance(other, PackageSet):
+            if self.packages == other.packages:
+                if self.label != other.label:
+                    raise ParamError("Different package set tags ('%s', '%s') refer to the same set of packages. " %
+                                     (self.label, other.label) + "Check if java attributes are different.")
```

Comment:
Because lists of packages are used to map to a label in the e-mail/html reporting. If you've got two different sets of packages using the same label, it's going to make the reporting a mess.